sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/SupportsDelete.java

```diff
@@ -28,6 +28,25 @@
  */
 @Evolving
 public interface SupportsDelete {
+
+  /**
+   * Checks whether it is possible to delete data from a data source table that matches filter
+   * expressions.
+   * <p>
+   * Rows should be deleted from the data source iff all of the filter expressions match.
+   * That is, the expressions must be interpreted as a set of filters that are ANDed together.
+   * <p>
+   * Spark will call this method to check if the delete is possible without significant effort.
+   * Otherwise, Spark will try to rewrite the delete operation and produce row-level changes
+   * if the data source table supports deleting individual records.
+   *
+   * @param filters filter expressions, used to select rows to delete when all expressions match
+   * @return true if the delete operation can be performed
+   */
+  default boolean canDeleteWhere(Filter[] filters) {
+    return true;
+  }
+
   /**
    * Delete data from a data source table that matches filter expressions.
```

Comment:
Wait .. so does this basically for failing fast? What's the diff from throwing an exception `IllegalArgumentException` as documented here? We can still check first and fail fast as documented in this API.

Reply:
Yea I have the same question.

Reply:
I tried to elaborate on this [here](https://github.com/apache/spark/pull/30562#discussion_r533990021). Let me know if that makes sense, @HyukjinKwon @cloud-fan.